src/proxy/outbound.rs

```diff
@@ -369,28 +379,27 @@ impl OutboundConnection {
 
         let us = us.unwrap();
         // For case upstream server has enabled waypoint
-        if !us.workload.waypoint_addresses.is_empty() {
-            let waypoint_address = us.workload.choose_waypoint_address().unwrap();
-            // Even in this case, we are picking a single upstream pod and deciding if it has a remote proxy.
-            // Typically this is all or nothing, but if not we should probably send to remote proxy if *any* upstream has one.
-            let waypoint_workload = match self.pi.workloads.fetch_workload(&waypoint_address).await
-            {
-                Some(wl) => wl,
-                None => return Err(Error::UnknownWaypoint(waypoint_address, downstream)),
-            };
-            return Ok(Request {
-                // Always use HBONE here
-                protocol: Protocol::HBONE,
-                source: source_workload,
-                // Use the original VIP, not translated
-                destination: target,
-                destination_workload: Some(us.workload),
-                expected_identity: Some(waypoint_workload.identity()),
-                gateway: SocketAddr::from((waypoint_address, 15008)),
-                // Let the client remote know we are on the inbound path.
-                direction: Direction::Inbound,
-                request_type: RequestType::ToServerWaypoint,
-            });
+        match self.pi.workloads.find_waypoint(us.workload.clone()).await {
+            Ok(None) => {} // do nothing
```

Comment:
**suggestion:** should we add some sort of debug log entry here so it's more straightforward to understand that we landed in this arm of the match statement?

Reply:
Isn't it checking whether it has waypoint? If not, should drop

Reply:
SO I understand with it is now, if the waypoint has no workload found, it simply send the traffic to dest workload without through waypoint, right?

Reply:
If so before a waypoint running up, the request to upstream may skip some auth policy

Reply:
> if the waypoint has no workload found, it simply send the traffic to dest workload without through waypoint, right?

correct

> If so before a waypoint running up, the request to upstream may skip some auth policy

no; the contract for `find_waypoint(workload)` is that if the workload is configured with a waypoint service, then it either finds a backing workload for that service or returns an error. we only return `Ok(None)` if the workload itself does not have a waypoint at all